terratorch/tasks/wxc_downscaling_task.py

```diff
@@ -97,20 +102,48 @@ def instantiate_metrics():
         self.test_metrics = MetricCollection(instantiate_metrics(), prefix="test/")
 
     def training_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> Tensor:
-        raise NotImplementedError("This function is not yet implemented.")
+
+        x = batch["image"]
+        mask = batch["mask"]
+        model_output: ModelOutput = self(x)
+        y = mask
+        loss = self.train_loss_handler.compute_loss(model_output, y, self.criterion, None)
+        self.train_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=y.shape[0])
+        y_hat = model_output.output
+        self.train_metrics(y_hat, y)
+        self.log_dict(self.train_metrics, on_epoch=True)
+
+        return loss["loss"]
 
     def validation_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> None:
-        raise NotImplementedError("This function is not yet implemented.")
+        """Compute the train loss and additional metrics.
+
+        Args:
+            batch: The output of your DataLoader.
+            batch_idx: Integer displaying index of this batch.
+            dataloader_idx: Index of the current dataloader.
+        """
+        x = batch["image"]
+        y = batch["mask"]
+        model_output: ModelOutput = self(x)
+
+        loss = self.train_loss_handler.compute_loss(model_output, y, self.criterion, None)
+        self.train_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=y.shape[0])
+        y_hat = model_output.output
+        self.train_metrics(y_hat, y)
+        self.log_dict(self.train_metrics, on_epoch=True)
+
+        return loss["loss"]
 
     def test_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> None:
         raise NotImplementedError("This function is not yet implemented.")
```

Comment:
this should be pretty straightforward to add, given the others are already there

Reply:
I'm not sure about this comment.